token/program-2022/src/extension/permanent_delegate.rs

```diff
@@ -16,3 +17,14 @@ pub struct PermanentDelegate {
 impl Extension for PermanentDelegate {
     const TYPE: ExtensionType = ExtensionType::PermanentDelegate;
 }
+
+/// Attempts to get the permanent delegate from the TLV data, returning None
+/// if the extension is not found
+pub fn maybe_get_permanent_delegate<S: BaseState, BSE: BaseStateWithExtensions<S>>(
```

Comment:
Bikeshedding: either `maybe_permanent_delegate` or `get_permanent_delegate` would seem more in line with precedent, at least in the monorepo where I'm more familiar. Looks like we don't really have other methods with `maybe` prefixes in SPL repo.